aiida/orm/autogroup.py

```diff
@@ -7,173 +7,286 @@
 
-        self.include = the_include_classes
+        :param include: a list of valid entry point strings (one of which could be the string 'all')
+        """
+        if isinstance(include, str):
+            include = [include]
+        self.validate(include)
+        if include is not None and self.get_exclude() is not None:
+            # It's ok to set None, both as a default, or to 'undo' the include list
+            raise exceptions.ValidationError('Cannot both specify exclude and include')
+        self._include = include
 
-    def set_include_with_subclasses(self, include):
+    def set_group_label_prefix(self, label_prefix):
         """
-        Set the list of classes to include in the autogrouping.
-        Will also include their derived subclasses.
+        Set the label of the group to be created
         """
-        the_include_classes = self._validate(include, is_exact=False)
-        self.include_with_subclasses = the_include_classes
+        if not isinstance(label_prefix, str):
+            raise exceptions.ValidationError('group label must be a string')
+        self._group_label_prefix = label_prefix
 
     def set_group_name(self, gname):
+        """Set the name of the group.
+
+        .. deprecated:: 1.2.0
+            Will be removed in `v2.0.0`, use :py:meth:`.set_group_label_prefix` instead.
         """
-        Set the name of the group to be created
+        warnings.warn('function is deprecated, use `set_group_label_prefix` instead', AiidaDeprecationWarning)  # pylint: disable=no-member
+        return self.set_group_label_prefix(label_prefix=gname)
+
+    @staticmethod
+    def _matches(string, filter_string):
+        """Check if 'string' matches the 'filter_string' (used for include and exclude filters).
+
+        If 'filter_string' does not end with a % sign, perform an exact match.
+        Otherwise, strip the '%' sign and match with string.startswith(filter_string[:-1]).
+
+        :param string: the string to match.
+        :param filter_string: the filter string.
         """
-        if not isinstance(gname, str):
-            raise exceptions.ValidationError('group name must be a string')
-        self.group_name = gname
+        if filter_string.endswith('%'):
+            return string.startswith(filter_string[:-1])
+        return string == filter_string
 
-    def is_to_be_grouped(self, the_class):
+    def is_to_be_grouped(self, node):
         """
-        Return whether the given class has to be included in the autogroup according to include/exclude list
+        Return whether the given node has to be included in the autogroup according to include/exclude list
 
-        :return (bool): True if the_class is to be included in the autogroup
+        :return (bool): True if ``node`` is to be included in the autogroup
         """
+        # I import here to avoid circular imports
+        from aiida.orm.nodes.process import ProcessNode
+
+        # strings, including possibly 'all'
```

Comment:
this comment is no longer true, correct?

Reply:
Correct. Removed.